src/compiler/parse.rs

```diff
@@ -180,6 +180,7 @@ impl Parser {
             Token::Sub => self.sub(left),
             Token::Mul => self.mul(left),
             Token::Div => self.div(left),
+            Token::Percent => self.remainder(left),
```

Comment:
Would it make more sense to call it `Token::Rem` and the function `self.rem(...)` for consistency with the other operators?

Reply:
I kept it as `Percent` in the lexing stage because at that point it's really just a symbol, it gets the meaning of remainder only later, at the parsing stage actually TBH, only at the FFI stage, but ATM we don't use it for anything else, I actually considered changing it to `Percent` in the parsing stage as well, exactly for these reasons, because technically, one could decide to use '%' as an operator for other functions, on other types, but that would be getting ahead of ourselves.

Anyways, the point is I thought it makes sense to differentiate the symbol name when lexing with it's semantic meaning when parsing.

I could change it, your call, let me know.

Reply:
I don't really think there are any other cases where we use `%` in the language itself. I think the case could be made for something like `-`, because that can be subtraction or unary negation, but `%` itself is pretty much only used for remainder from what I've seen.

Reply:
I'm mostly using python for work these days, so I see many weird overloads for operators.
a pretty common is using '%' for custom string formatting, if someone won't like the default option (like python, ruby, and probably more).
Of course you could always argue that one _shouldn't_ do that, and that a normal function is better for this use case, and that '%' should be used only as the `remainder` operator, I guess since operators are actually implemented with macros + FFI, it sounds pretty common for me that someone would want to override it for some DSL or something, but maybe I'm already thinking too far ahead...?
Personally I like keeping my options open, but I get that it could also lead to a mess, so IDK.

Reply:
Here are my thoughts: I think that % string formatting isn't ideal, and much prefer the f-string formatting present in languages like Python, and Rust. I think that we'll eventually support overloading for all operators, but this won't be done though implementing something for `%` directly, rather you'd have to implement a `rem` trait, something like `__rem__` in Python (actual traits though, not just dunder methods on classes.)

Reply:
Ok, I have a great ability to over-discuss the _too small_ details, and I feel like we're getting to that point so I'll try to avoid it.
I think we're not at the point where we can accurately know how to work on the things I brought up here, traits sound better and safer to me, but we have a few bridges to cross before actually getting there, right?
So bottom line:
I like the separation in names between the token (percent) and it's semantic meaning (remainder), regardless of whether it will be overloaded or not, I like it like this, I don't find it confusing, I actually think it's clearer, but if you don't agree, I'll change it, it's not a big deal.

Reply:
Thank you. If it ever becomes clear that `%` will be used for things other than remainder going forward, we can change the name of the token variant, no biggie 😄